monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -146,6 +152,16 @@ object Local {
        try { $f } finally { $Local.setContext($saved) }
        """)
     }
+
+    def localLetCurrent(f: Tree): Tree = {
+      val saved = util.name("saved")
+      val Local = symbolOf[Local[_]].companion
+      resetTree(
+        q"""
+       val $saved = $Local.getContext()
+       try { $f } finally { $Local.setContext($saved) }
```

Comment:
I can add another parameter to the macro, so that if...else happens here. This would let me make things more DRY but the `bindCurrent` signature will seem to be too special-cased, so I'll probably have to make it `private[monix]`